prov/gni/src/gnix.h

```diff
@@ -59,10 +60,219 @@ extern "C"
 #include <fi_indexer.h>
 #include <fi_rbuf.h>
 #include <fi_list.h>
+#include "gni_pub.h"
+#include "ccan/list.h"
 
 #define GNI_MAJOR_VERSION 0
 #define GNI_MINOR_VERSION 5
 
+/*
+ * useful macros
+ */
+
+#ifndef likely
+#define likely(x)   __builtin_expect((x),1)
+#endif
+#ifndef unlikely
+#define unlikely(x) __builtin_expect((x),0)
+#endif
+
+#ifndef FLOOR
+#define FLOOR(a,b)      ((long long)(a) - ( ((long long)(a)) %(b)))
+#endif
+
+#ifndef CEILING
+#define CEILING(a,b)    ((long long)(a) <= 0LL ? 0 :  (FLOOR((a)-1,b) + (b)))
+#endif
+
+#ifndef IN
```

Comment:
nothing yet but they are used in some of the code we may recycle for code which hasn't appeared yet.